sundials/invocation_context.h is missing an include.

@vovannikov We compile each header as a separate file as part of the test suite to see whether everything that's used in the header is also `#include`d in that header. We have the following error:
```
In file included from /scratch/users/testsuite/testsuite-WlN4kfZV/dealii/tests/all-headers/test_header.cc:14:
/scratch/users/testsuite/testsuite-WlN4kfZV/dealii/include/deal.II/sundials/invocation_context.h:48:7: error: 'SUNContext' does not name a type
   48 |       SUNContext &arkode_ctx;
      |       ^~~~~~~~~~
gmake[3]: *** [CMakeFiles/sundials-invocation_context.h.release.dir/build.make:80: CMakeFiles/sundials-invocation_context.h.release.dir/test_header.cc.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:72817: CMakeFiles/sundials-invocation_context.h.release.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:72850: CMakeFiles/sundials-invocation_context.h.release.build.dir/rule] Error 2
gmake: *** [Makefile:35156: sundials-invocation_context.h.release.build] Error 2
Test all-headers/sundials/invocation_context.h.release: BUILD
```
Can you see what header needs to be `#include`d to address this error?

The full error is here: https://cdash.dealii.org/tests/7183962. This is part of #19608.

--- a/include/deal.II/sundials/invocation_context.h
+++ b/include/deal.II/sundials/invocation_context.h
@@ -21,7 +21,10 @@
 
 #ifdef DEAL_II_WITH_SUNDIALS
 
-#  include <sundials/sundials_types.h>
+#  if DEAL_II_SUNDIALS_VERSION_GTE(6, 0, 0)
+#    include <sundials/sundials_context.h>
+#    include <sundials/sundials_types.h>
+#  endif
 
 #  include <exception>
 

